ext/debug/segger/systemview/SEGGER_SYSVIEW_Conf.h

```diff
@@ -1,11 +1,25 @@
 #pragma once
 
 #include <stdint.h>
+#include <irq.h>
 
 #define SEGGER_SYSVIEW_GET_TIMESTAMP sysview_get_timestamp
 #define SEGGER_SYSVIEW_GET_INTERRUPT_ID sysview_get_interrupt
 
 uint32_t sysview_get_timestamp(void);
 uint32_t sysview_get_interrupt(void);
+extern unsigned int sysview_irq_key;
+extern size_t sysview_irq_cnt;
 
-#define SEGGER_SYSVIEW_RTT_BUFFER_SIZE 4096
+#define SEGGER_SYSVIEW_RTT_BUFFER_SIZE CONFIG_SEGGER_SYSVIEW_RTT_BUFFER_SIZE
+
+// Lock SystemView (nestable)
+#define SEGGER_SYSVIEW_LOCK()		sysview_irq_cnt++;		       \
```

Comment:
1. Possible race between incrementation and the lock.
2. Hidden errors when using this macro after loop or if.
3. Hidden access to global variables.
4. No need of the macro - the irq_lock() could be nested.